src/v3/src/transformer/transaction/transform.ts

```diff
@@ -50,6 +61,54 @@ const transformRemediationNameForIdp: TransformStepFnWithOptions = (options) =>
   return formbag;
 };
 
+/**
+ * To support `idps` configuration in OIE.  Adds IDP buttons from widget config.
+ * https://github.com/okta/okta-signin-widget#openid-connect
+ */
+const injectIdPConfigButtonToRemediation: TransformStepFnWithOptions = (options) => (formbag) => {
+  const { transaction, widgetProps } = options;
+  const widgetRemediations = transaction.neededToProceed;
+  const hasIdentifyRemediation = widgetRemediations.some((r) => r.name === IDX_STEP.IDENTIFY);
+  if (!hasIdentifyRemediation) {
+    return formbag;
+  }
+
+  // @ts-expect-error OKTA-609461 - idps missing from WidgetOptions type
+  const idpsConfig = widgetProps.idps;
+  if (Array.isArray(idpsConfig)) {
+    const existsRedirectIdpIds: Record<string, boolean> = {};
+    widgetRemediations.forEach((r) => {
+      if (r.name === IDX_STEP.REDIRECT_IDP && r.idp) {
+        existsRedirectIdpIds[r.idp.id] = true;
+      }
+    });
+    const { baseUrl } = widgetProps;
+    const { stateHandle } = transaction.context;
+    const redirectIdpRemediations = idpsConfig
+      .filter((c) => !existsRedirectIdpIds[c.id]) // omit idps that are already in remediation.
+      .map((idpConfig) => {
+        const idp = {
+          id: idpConfig.id,
+          name: idpConfig.text,
+        };
+        const redirectUri = `${baseUrl}/sso/idps/${idpConfig.id}?stateToken=${stateHandle}`;
+        if (idpConfig.className) {
+          // @ts-expect-error OKTA-609464 - className missing from IdpConfig type
+          idp.className = idpConfig.className;
+        }
+        return {
+          name: IDX_STEP.REDIRECT_IDP,
+          type: idpConfig.type,
+          idp,
+          href: redirectUri,
+        };
+      });
+    transaction.neededToProceed = widgetRemediations.concat(redirectIdpRemediations);
```

Comment:
Usually (specifically for Gen 3), any modifications done to the `neededToProceed` property in the transaction, should also be replicated to the `nextStep` and `availableSteps` property of the transaction object. The reason for this is because _most_ of the transformers in Gen 3, rely heavily on the `nextSTep` or `availableSteps` properties and only in rare cases do we ignore those and go directly into the neededToProceed property.

Reply:
yeah this whole function for `openid-connect` was pretty much ported over quickly completely from v2 and doesnt really fit as well in v3 context.  Im going to see if I can extract this feature out of here in a way that wont need to modify the transaction at all.